_posts/2019-03-14-2019-kafka-connector.md

```diff
@@ -0,0 +1,217 @@
+```
+
+In the second terminal follow the Kafka Connector logs with this command:
+
+```bash
+$ kubectl logs $(kubectl get pods -o name -n openfaas | grep -m1 kafka-connector | cut -d'/' -f 2) -n openfaas --tail 100 -f
+```
+
+In the first terminal, inside the producer pod shell session invoke the producer binary with those command line arguments:
+
+```bash
+/usr/bin producer -messages=1 -broker=kafka:9092 -pause 1s -topic=faas-request
+```
+
+The binary provides default message `Test the function.` You should see the output of the function in the kafka-connector logs:
+```
+...
+2019/03/16 10:08:26 Syncing topic map
+[#1] Received on [faas-request,0]: 'Test the function.'
+2019/03/16 10:08:26 Invoke function: figlet
+2019/03/16 10:08:27 connector-sdk got result: [200] faas-request => figlet (480) bytes
+[200] faas-request => figlet
+ _____         _     _   _             __                  _   _               
+|_   _|__  ___| |_  | |_| |__   ___   / _|_   _ _ __   ___| |_(_) ___  _ __    
+  | |/ _ \/ __| __| | __| '_ \ / _ \ | |_| | | | '_ \ / __| __| |/ _ \| '_ \   
+  | |  __/\__ \ |_  | |_| | | |  __/ |  _| |_| | | | | (__| |_| | (_) | | | |_ 
+  |_|\___||___/\__|  \__|_| |_|\___| |_|  \__,_|_| |_|\___|\__|_|\___/|_| |_(_)
+                                                                               
+
+2019/03/16 10:08:29 Syncing topic map
+2019/03/16 10:08:32 Syncing topic map
+2019/03/16 10:08:35 Syncing topic map
+...
+```
+
+## But why do I need this combination?
+
+There are multiple real world examples where Kafka in combination with the Serverless approach provided by OpenFaaS becomes really powerful tool as Kafka can make even the smallest of systems extensible and OpenFaaS provides us with the quick way to add those extensions.
+
+Lets say you need to process customer information. When customer applies for you application you need to process it and store the information in a database and there are tons of customers. Just create topic for example `user-creation` which will be populated with the information and then create OpenFaaS function which handles the processing and updating of that information to the database.
+
+If we extend on the example above, again you would like to add some quick feature for the user.Attach another function to the same topic to extend the already existing functionality or just add another topic and write the function which handles that feature, it is as simple as that. 
+
+## Wrapping up
+
+Kafka Connector implements the [Connector SDK](https://github.com/openfaas-incubator/connector-sdk) with which you can create your own invoking mechanism. 
+
+You can check the existing connectors like the [Cron Connector](https://github.com/zeerorg/cron-connector) or [vCenter Connector](https://github.com/openfaas-incubator/vcenter-connector).
+
+Now you are ready to attach the Kafka Connector plugin to your own Kafka deployment.
```

Comment:
Final call to action.. checkout my latest posts for an example.

Reply:
Prompted to extend the Hello world example and try multiple topics ✔️